checkov/kustomize/runner.py

```diff
@@ -380,30 +375,35 @@ def run(self, root_folder, external_checks_dir=None, files=None, runner_filter=R
 
         return report
 
-    def _curWriterRenameAndClose(self, cur_writer, FilePath):
+    def _curWriterValidateStoreMapAndClose(self, cur_writer, FilePath):
         currentFileName = cur_writer.name
         cur_writer.close()
         # Now we have a complete k8s manifest as we closed the writer, and it's temporary file name (currentFileName) plus the original file templated out (FilePath)
         # Rename them to useful information from the K8S metadata before conting.
-        # Then keep a mapping of temp files to original repo locations for use with Checkov output later.
+        # Then keep a mapping of template files to original kustomize repo locations for use with Checkov output later.
         try:
             with open(currentFileName) as f:
                 currentYamlObject = yaml.safe_load(f)
-                itemName = []
-                itemName.append(currentYamlObject['kind'])
-                if 'namespace' in currentYamlObject['metadata']:
-                    itemName.append(currentYamlObject['metadata']['namespace'])
-                else:
-                    itemName.append("default")
-                if 'name' in currentYamlObject['metadata']:
-                    itemName.append(currentYamlObject['metadata']['name'])
+                # Validate we have a K8S manifest
+                if "apiVersion" in currentYamlObject:
+                    itemName = []
+                    itemName.append(currentYamlObject['kind'])
+                    if 'namespace' in currentYamlObject['metadata']:
+                        itemName.append(currentYamlObject['metadata']['namespace'])
+                    else:
+                        itemName.append("default")
+                    if 'name' in currentYamlObject['metadata']:
+                        itemName.append(currentYamlObject['metadata']['name'])
+                    else:
+                        itemName.append("noname")
+            
+                    filename = f"{'-'.join(itemName)}.yaml"
+                    newFullPathFilename = str(pathlib.Path(currentFileName).parent) + "/" + filename
```

Comment:
```suggestion
                    newFullPathFilename = str(pathlib.Path(currentFileName).parent / filename) 
```
when you use `pathlib.Path` you can contatence paths by simply writing `... / ...` this will also make the correct path for Windows systems.